File: dynamic-programming/pascals-triangle-ii.cpp
class Solution {
public:
    vector<int> getRow(int n) {
        vector <vector<int>> ps;

        vector<int> re;
        for(int i=0;i<=n;i++){
            vector<int> row(i+1,1);
            for(int j=1;j<i;j++){
                row[j]=ps[i-1][j-1]+ps[i-1][j];
            }
            ps.push_back(row);
            re=row;
        }

        return re;
    }
};

File: 119-pascals-triangle-ii/pascals-triangle-ii.java
class Solution {
    public List<Integer> getRow(int rowIndex) {
        int row=0;
        ArrayList<ArrayList<Integer>> ans=new ArrayList<>();
        for(int i=0;i<=rowIndex;i++){
            ArrayList<Integer> level=new ArrayList<>();
            for(int j=0;j<=i;j++){
                if(j==0||j==i){
                    level.add(1);
                }else{
                    int val=ans.get(i-1).get(j-1)+ans.get(i-1).get(j);
                    level.add(val);
                }
            }
            ans.add(level);
            
        }

        return ans.get(rowIndex);
    }
}